scopes/component/ui/version-dropdown/version-dropdown-placeholder.module.scss

```diff
@@ -4,94 +4,109 @@
   border-radius: 6px;
   transition: background-color 300ms ease-in-out;
   height: 30px;
-  padding: 0 8px;
   display: flex;
   align-items: center;
   box-sizing: border-box;
   user-select: none;
+  flex-wrap: nowrap;
+  gap: 8px;
+  padding: 8px;
 
-  > div {
-    margin-right: 5px;
-  }
   &:hover {
     background-color: var(--bit-bg-heavy);
   }
 
-  > div {
-    max-width: 80px;
-  }
   &.disabled {
     cursor: default;
     background-color: #ededed;
     > span {
       display: none;
     }
   }
+
   .versionName {
-    &.tag {
-      min-width: fit-content;
-    }
-    &.snap {
-      min-width: 20%;
-    }
+    flex-grow: 1;
+    white-space: nowrap;
+    overflow: hidden;
+    text-overflow: ellipsis;
+    min-width: fit-content;
+  }
+
+  .commitMessage {
+    flex-grow: 1;
+    overflow: hidden;
+    white-space: nowrap;
+    text-overflow: ellipsis;
+  }
+
+  .versionTimestamp {
+    white-space: nowrap;
+    overflow: hidden;
+    text-overflow: ellipsis;
+  }
+
+  .versionUserAvatar {
+    flex: none;
+    margin-right: 5px;
   }
 }
 
 .detailed {
   display: flex;
   align-items: center;
   box-sizing: border-box;
-  padding: 0 8px;
   height: 30px;
   border-radius: 6px;
   user-select: none;
   transition: background-color 300ms ease-in-out;
   border: 1px solid var(--bit-border-color, #babec9);
-
   cursor: pointer;
-  > div {
-    margin-right: 5px;
-  }
+  flex-wrap: nowrap;
+  gap: 8px;
+  padding: 8px;
+
   &:hover {
     background-color: var(--bit-bg-heavy);
   }
-  > div {
-    max-width: 80px;
-  }
+
   &.disabled {
     cursor: default;
     background-color: #ededed;
     > span {
       display: none;
     }
   }
+
   .versionName {
-    &.tag {
-      min-width: fit-content;
-    }
-    &.snap {
-      min-width: 20%;
-    }
-  }
-  .commitMessage,
-  .emptyMessage {
-    max-width: 50%;
-    padding: 0px 8px;
-  }
-  .emptyMessage {
-    font-style: italic;
-    color: var(--bit-text-color-light, #6c707c);
-  }
-  .author {
-    text-transform: capitalize;
+    flex-grow: 1;
+    white-space: nowrap;
+    overflow: hidden;
+    text-overflow: ellipsis;
     min-width: fit-content;
   }
+
+  .commitMessage {
+    flex-grow: 1;
+    overflow: hidden;
+    white-space: nowrap;
+    text-overflow: ellipsis;
+  }
+
   .versionTimestamp {
-    min-width: fit-content;
-    margin-right: 2px;
-    text-align: right;
+    white-space: nowrap;
+    overflow: hidden;
+    text-overflow: ellipsis;
```

Comment:
Consider using https://bit.cloud/teambit/design/ui/styles/ellipsis
This component have both Component and Class only exported.